sdk/storage/azure-storage-blob/src/main/java/com/azure/android/storage/blob/transfer/TransferClient.java

```diff
@@ -3,93 +3,274 @@
+            this.networkType = networkType;
+            return this;
+        }
+
+        /**
+         * Sets whether device battery should be at an acceptable level for running the transfers.
+         * The default value is {@code false}.
+         *
+         * @param requiresBatteryNotLow {@code true} if the battery should be at an acceptable level
+         *                              for the transfers to run
+         * @return Builder with provided battery requirement set
+         */
+        public Builder setRequiresBatteryNotLow(boolean requiresBatteryNotLow) {
+            this.requiresBatteryNotLow = requiresBatteryNotLow;
+            return this;
+        }
+
+        /**
+         * Sets whether the device's available storage should be at an acceptable level for running
+         * the transfers. The default value is {@code false}.
+         *
+         * @param requiresStorageNotLow {@code true} if the available storage should not be below a
+         *                              a critical threshold for the transfer to run
+         * @return Builder with provided storage requirement set
+         */
+        public Builder setRequiresStorageNotLow(boolean requiresStorageNotLow) {
+            this.requiresStorageNotLow = requiresStorageNotLow;
+            return this;
+        }
+
+        /**
+         * Set the {@link Executor} used by {@link TransferClient} for all its internal
+         * book keeping, which includes creating DB entries for transfer workers, querying DB
+         * for status, submitting transfer request to {@link WorkManager}.
+         *
+         * TransferClient will enqueue maximum of two command to the serialBackgroundExecutor at any time.
+         *
+         * @param executor the serialBackgroundExecutor
+         * @return Builder with provided serialBackgroundExecutor set
+         */
+        public Builder setTaskExecutor(@NonNull Executor executor) {
+            this.serialBackgroundExecutor = new SerialExecutor(executor);
+            return this;
+        }
+
+        /**
+         * @return A {@link TransferClient} configured with settings applied through this builder
+         */
+        // Following annotation is added to suppress library restricted WorkManager
+        // androidx.work.Configuration Object access
+        @SuppressLint("RestrictedApi")
+        public TransferClient build() {
+            if (this.storageBlobClient == null) {
+                throw new IllegalArgumentException("storageBlobClient must be set.");
+            }
+            final Constraints.Builder constraintsBuilder = new Constraints.Builder();
+            constraintsBuilder.setRequiresCharging(this.requiresCharging);
+            constraintsBuilder.setRequiresDeviceIdle(this.requiresDeviceIdle);
+            if (this.networkType == null) {
+                throw new IllegalArgumentException("networkType must be set.");
```

Comment:
Maybe we should just default this to `CONNECTED` if not specified. That seems like a reasonable default.

Reply:
hmm, I thought of that a bit worried about it i.e. using a default that can "incur charge/money".

Reply:
Definitely a good thought. I looked at what WorkManager itself does, and if no Constraints are supplied, the default value is `Constraints.NONE`: https://android.googlesource.com/platform/frameworks/support/+/androidx-master-dev/work/workmanager/src/main/java/androidx/work/impl/model/WorkSpec.java#97

I'd feel okay doing the same, for consistency. Though I do totally agree that the workmanager is a general framework, while we're using it specifically for network transfers and so the appropriate default might be different for our use case. Wondering if @adrianhall has an opinion here?

Reply:
Just checked what AWS is doing in this case, aws-lib uses [ANY](https://github.com/aws-amplify/aws-sdk-android/blob/0a4eac920163f80b0e46fe93721a2cc93848f36d/aws-android-sdk-s3/src/main/java/com/amazonaws/mobileconnectors/s3/transferutility/TransferUtilityOptions.java#L184), equivalent that in WM is `CONNECTED`.

Reply:
Created an issue here: https://github.com/Azure/azure-sdk-for-android/issues/213.

Reply:
Based on the response to the issue, we will go with `CONNECTED` as default which is already specified in the code. Marking it resolved.